crates/sui-core/src/authority_aggregator.rs

```diff
@@ -2199,6 +2219,55 @@ where
         .await
     }
 
+    /// This function tries to get SignedTransaction OR CertifiedTransaction from
+    /// an given list of validators who are supposed to know about it.
+    pub async fn handle_transaction_info_request_from_some_validators(
+        &self,
+        tx_digest: &TransactionDigest,
+        // authorities known to have the transaction info we are requesting.
+        validators: &BTreeSet<AuthorityName>,
+        timeout_total: Option<Duration>,
+    ) -> SuiResult<(
+        Option<VerifiedSignedTransaction>,
+        Option<VerifiedCertificate>,
+    )> {
+        self.quorum_once_with_timeout(
+            None,
+            Some(validators),
+            |authority, client| {
+                Box::pin(async move {
+                    let response = client
+                        .handle_transaction_info_request(TransactionInfoRequest {
+                            transaction_digest: *tx_digest,
+                        })
+                        .await?;
+                    if let Some(certified_transaction) = response.certified_transaction {
+                        return Ok((None, Some(certified_transaction)));
+                    }
+
+                    if let Some(signed_transaction) = response.signed_transaction {
+                        return Ok((Some(signed_transaction), None));
+                    }
+
+                    // This validator could not give the transaction info, but it is supposed to know about the transaction.
+                    // This could also happen on epoch change boundary.
+                    warn!(name=?authority, ?tx_digest, "Validator failed to give info about a transaction, it's either byzantine or just went through an epoch change");
+                    Err(SuiError::ByzantineAuthoritySuspicion {
+                        authority,
+                        reason: format!(
+                            "Validator claimed to know about tx {:?} but did not return it when queried",
+                            tx_digest,
+                        )
+                    })
+                })
+            },
+            Duration::from_secs(2),
```

Comment:
Is this timer selected based on the Narwhal timeout delay (which is also 2 seconds)?